petastorm/reader.py

```diff
@@ -101,6 +101,9 @@ def __init__(self, dataset_url, schema_fields=None, shuffle=None, predicate=None
         #    c. partition: used to get a subset of data for distributed training
         # 4. Create a rowgroup ventilator object
         # 5. Start workers pool
+        if dataset_url is None or not isinstance(dataset_url, str):
```

Comment:
Shouldn't it be a `basestring` (to support unicode strings)?

Reply:
Ok, `basestring` is not python3. Not sure what is the fix then, but I just saw someone calling `Reader` with a unicode path from python2.7....

Reply:
Found this is a python2,3 compatible solution:
```python
isinstance(self._dataset_url, six.string_types)
```

Reply:
Ah, yeah thats better